src/select/multiselect/OptionGroupRenderer.js

```diff
@@ -3,9 +3,17 @@ import React from 'react'
 import PropTypes from 'prop-types'
 
 const OptionGroupRenderer = (props) => {
-  const { groupTitleKey, groupValueKey, option, options, value } = props
+  const {
+    groupTitleKey,
+    groupValueKey,
+    option,
+    options,
+    unfilteredOptions,
+    value,
+    valueKey } = props
+  const opts = unfilteredOptions.length > 0 ? unfilteredOptions : options
   const checked = _.some(value, (val) => {
-    const valueObj = props.getSelectValue({ options, groupValueKey, value: val })
+    const valueObj = props.getSelectValue({ options: opts, groupValueKey, value: val, valueKey })
```

Comment:
So it looks like here `OptionGroupRenderer` and `OptionRenderer` the primary change we're making is to ensure unfilteredOptions is passed into `getSelectValue`, so we get an accurate `valueObj` back, correct? But I noticed we pass in a custom `getSelectValue` prop in FilterMultiSelect. We should be able to use `const { options } = this.props` there instead of needing `unfilteredOptions` here. That keeps the responsibility of maintaining an accurate options list with the High Order Component. `FilterSelect` does this to account for the same problem.

Reply:
Got it 👍